latch_config/config.py

```diff
@@ -51,17 +70,92 @@ class LoggingMode(str, Enum):
 T = TypeVar("T")
 
 
-def read_config(x: type[T], env_prefix: str = "") -> T:
+class _IsDataclass(Protocol):
+    __dataclass_fields__: ClassVar[dict[str, Field]]
+
+
+DC = TypeVar("DC", bound=_IsDataclass)
+
+
+class NoValue: ...
+
+
+class NestyError(Exception):
+    def __init__(self, msg: str, children: list[Self] | None = None):
+        self.msg = msg
+        self.children = children
+
+    def __str__(self):
+        if self.children is None:
+            return self.msg
+
+        res: list[str] = [self.msg]
+        for child in self.children:
+            res.append(indent(str(child), " -> "))
+
+        return "\n".join(res)
+
+
+def parse_config_val(t: type[T], v: Any) -> T:
```

Comment:
```suggestion
def parse_config_val(t: type[T], v: object) -> T:
```
`object` is closer to `unknown`